DigitalLearningSolutions.Data.Tests/Services/PasswordResetServiceTests.cs

```diff
@@ -1,67 +1,218 @@
+
+        [Test]
+        public void Trying_to_send_password_reset_sends_email()
+        {
+            // Given
+            var emailAddress = "recipient@example.com";
+            var adminUser = Builder<AdminUser>.CreateNew().With(user => user.EmailAddress = emailAddress)
+                .Build();
+
+            A.CallTo(() => userService.GetUsersByEmailAddress(A<string>._))
+                .Returns((new[] { adminUser }.ToList(), new List<DelegateUser>()));
+
+            // When
+            passwordResetService.GenerateAndSendPasswordResetLink(emailAddress, "example.com");
+
+            // Then
+            A.CallTo(
+                    () =>
+                        emailService.SendEmail(
+                            A<Email>.That.Matches(
+                                e =>
+                                    e.To[0] == emailAddress &&
+                                    e.Cc.IsNullOrEmpty() &&
+                                    e.Bcc.IsNullOrEmpty() &&
+                                    e.Subject == "Digital Learning Solutions Tracking System Password Reset"))
+                )
+                .MustHaveHappened();
+            ;
+        }
+
+        [Test]
+        public async Task Reset_password_hash_is_invalid_if_125_minutes_old()
+        {
+            // Given
+            var createTime = DateTime.UtcNow;
+
+            var resetPasswordModel = Builder<ResetPassword>.CreateNew()
+                .With(m => m.Id = 1)
+                .With(m => m.PasswordResetDateTime = createTime)
+                .With(m => m.ResetPasswordHash = "Old Hash Brown")
+                .Build();
+            A.CallTo(() => passwordResetDataService.FindAsync(1)).Returns(Task.FromResult(resetPasswordModel));
+
+            var candidate = Builder<DelegateUser>.CreateNew()
+                .With(user => user.ResetPasswordId = resetPasswordModel.Id)
+                .Build();
+            A.CallTo(() => userService.GetUsersByEmailAddress(A<string>._))
+                .Returns((new List<AdminUser>(), new[] { candidate }.ToList()));
+
+            GivenCurrentTimeIs(createTime.Add(TimeSpan.FromMinutes(125)));
+
+            // When
+            var isValid = await passwordResetService.PasswordResetHashIsValidAsync("email", "New Hash Brown");
+
+            // Then
+            isValid.Should().BeFalse();
+        }
+
+        [Test]
+        public async Task New_reset_password_is_valid_115_minutes_later()
+        {
+            // Given
+            var createTime = DateTime.UtcNow;
+
+            var resetPasswordModel = Builder<ResetPassword>.CreateNew()
+                .With(m => m.Id = 1)
+                .With(m => m.PasswordResetDateTime = createTime)
+                .With(m => m.ResetPasswordHash = "New Hash Brown")
+                .Build();
+            GivenResetPasswordExists(resetPasswordModel);
+
+            var candidate = Builder<DelegateUser>.CreateNew().With(user => user.ResetPasswordId = resetPasswordModel.Id)

```

Comment:
Minor, but could we put the "With" on a new line too for better readability (like above)